text/0000-transaction-payload/0000-transaction-payload.md

```diff
@@ -0,0 +1,748 @@
+
+The bundle concept has proven to be tedious, spawning a plethora of problems:
+* Since the data making up the bundle is split across multiple vertices, it complicates the validation of the entire transfer. Instead of being able to immediately tell whether a bundle is valid or not, a node implementation must first collect all parts of the bundle before any actual validation can happen. This increases the complexity of the node implementation immensely.
+* Reattaching the tail transaction of a bundle causes the entire transfer to be reapplied.
+* Due to the split across multiple transaction vertices and having to do PoW for each of them, a created bundle might already be lazy in terms of where it attaches, reducing its chances to get confirmed.
+
+To fix the problems mentioned above and creating a more flexible transaction structure, we want to achieve a self-contained transaction structure defining the data of the entire transfer as a payload to be embedded into a message.
+
+The new transaction structure should fulfill following criteria:
+* Support for Ed25519 (and thus reusable addresses) and WOTS signatures.
+* Support for adding new types of signature schemes, addresses, inputs and outputs as part of protocol upgrades.
+* Self-contained as in being able to validate the transaction immediately after receiving it.
+* Enable unspent transaction outputs (UTXO) as inputs instead of an account based model. (UTXO enables easier double-spend detection)
+
+# Detailed design
+
+## UTXO
+
+The *unspent transaction output* (UTXO) model defines a ledger state where balances are not directly associated to addresses but to the outputs of transactions. In this model, transactions specify the outputs of previous transactions as inputs, which are consumed in order to create new outputs. A transaction must consume the entirety of the specified inputs.
+
+Using an UTXO based model provides several benefits:
+* Parallel validation of transactions.
+* Easier double-spend detection, since conflicting transactions would reference the same UTXO.
+* Replay-protection which is especially important when having reusable addresses. Replaying the same transaction would manifest itself as already being applied or existent and thus not have any impact.
+* Technically seen, balances may no longer be associated to addresses which raises the level of abstraction and thus enables other types of outputs. Consider for example a type of output which specifies the balance to be unlocked by a transaction which must fulfill a very hard Proof-of-Work difficulty or supply some other unlock criteria etc.
+
+Within a transaction using UTXOs, inputs and outputs make up the to be signed data of the transaction. The section unlocking the inputs is called *unlock block*. An unlock block may contain a signature proving ownership of a given input's address and/or other unlock criteria.
+
+The following image depicts the flow of funds using UTXO:
```

Comment:
Will there be one 'genesis' transaction for each address with balance at the time of convertion to new tx-layout ?

Reply:
This process will be described in a subsequent RFC.